cylc/flow/graph_parser.py

```diff
@@ -321,23 +323,46 @@ def parse_graph(self, graph_string: str) -> None:
         part_lines = []
         for i, _ in enumerate(non_blank_lines):
             this_line = non_blank_lines[i]
-            if i == 0 and this_line.startswith(self.__class__.ARROW):
-                # First line can't start with an arrow.
-                raise GraphParseError(f"Leading arrow: {this_line}")
+            for seq in self.CONTINUATION_STRS:
+                if i == 0 and this_line.startswith(seq):
+                    # First line can't start with an arrow.
+                    raise GraphParseError(f"Leading {seq}: {this_line}")
             try:
                 next_line = non_blank_lines[i + 1]
             except IndexError:
                 next_line = ''
-                if this_line.endswith(self.__class__.ARROW):
-                    # Last line can't end with an arrow.
-                    raise GraphParseError(
-                        f"Trailing arrow: {this_line}")
+                for seq in self.CONTINUATION_STRS:
+                    if this_line.endswith(seq):
+                        # Last line can't end with an arrow, & or |.
+                        raise GraphParseError(
+                            f"Dangling {seq}:"
+                            f"{this_line}"
+                        )
             part_lines.append(this_line)
+
+            # Check that a continuation sequence doesn't end this line and
+            # begin the next:
             if (
-                this_line.endswith(self.__class__.ARROW) or
-                next_line.startswith(self.__class__.ARROW)
+                this_line.endswith(self.CONTINUATION_STRS) and
+                next_line.startswith(self.CONTINUATION_STRS)
             ):
+                raise GraphParseError(
+                    'Consecutive lines end and start with continuation '
+                    'characters:\n'  # (not "sequence")
+                    f'{this_line}\n'
+                    f'{next_line}'
+                )
+
+            # Check that line ends with a valid continuation sequence:
+            if (any(
+                this_line.endswith(seq) or next_line.startswith(seq) for
```

Comment:
Use `endswith(tuple)` here too (and elsewhere in the file?) instead of iterating?

Reply:
COFFEE ERROR: Tim has not had coffee yet.

Reply:
On inspection I've decided to leave 
```
            for seq in self.CONTINUATION_STRS:
                if i == 0 and this_line.startswith(seq):
                    # First line can't start with an arrow.
                    raise GraphParseError(f"Leading {seq}: {this_line}")
```
Alone, because I think that providing the offending leading/dangling sequence to users is a worthwhile good. If you can see a nice way of doing it let me know. I tried seq = ternery expression, but this ended up being 5 lines long to fit our line length.

Reply:
Yep, no worries 👍🏻